allensdk/brain_observatory/vbn_2022/metadata_writer/lims_queries.py

```diff
@@ -0,0 +1,758 @@
+from typing import List, Tuple, Dict, Any, Optional
+import pandas as pd
+from allensdk.internal.api import PostgresQueryMixin
+
+from allensdk.internal.api.queries.lims_queries import (
+    behavior_sessions_from_ecephys_session_ids,
+    foraging_id_map_from_behavior_session_id,
+    _sanitize_uuid_list)
+
+from allensdk.internal.api.queries.mtrain_queries import (
+    session_stage_from_foraging_id)
+
+from allensdk.brain_observatory.vbn_2022.\
+    metadata_writer.dataframe_manipulations import (
+        _add_prior_omissions_to_behavior,
+        _add_prior_omissions_to_ecephys,
+        _add_session_number,
+        _add_age_in_days,
+        _patch_date_and_stage_from_pickle_file,
+        _add_experience_level,
+        _add_images_from_behavior)
+
+from allensdk.brain_observatory.behavior.behavior_project_cache.tables \
+    .util.prior_exposure_processing import (
+        get_image_set,
+        get_prior_exposures_to_image_set,
+        get_prior_exposures_to_session_type)
+
+
+def get_list_of_bad_probe_ids(
+        lims_connection: PostgresQueryMixin,
+        probes_to_skip: List[Dict[str, Any]]) -> List[int]:
+    """
+    Given a list of probes to skip,each of the form
+
+    {'session': ecephys_session_id,
+     'probe': probe_name}
+
+    return a list of the ecephys_probe_ids associated with
+    the bad probes.
+    """
+
+    where_clause = ""
+    for probe in probes_to_skip:
+        if len(where_clause) > 0:
+            where_clause += " OR "
+        where_clause += f"(ecephys_session_id={probe['session']}"
+        where_clause += f" AND name='{probe['probe']}')"
```

Comment:
I don't think that's what I want. This query is only grabbing probes where the pair `(ecephys_session_id, probe_name)` is satisfied. Grabbing all of the sessions in `<list of ecephys_session_ids>` would grab too many probes and grabbing all of the probes in `<list of probe names>` would grab too many sessions (i.e. if `(ecephys_session_id=1, probe_name='probeA')` is bad but `(ecephys_session_id=2, probe_name='probeA')` is good; I'm pretty sure probe names are not globally unique).

Reply:
I meant this
```
WHERE (ecephys_session_id=... AND name IN (...)) OR (...)
```

instead of 
```
WHERE (ecephys_session_id=... AND name=...) OR (...) 
```

but nevermind, it's not a worthwhile change.